### ISSUE TITLE
ssl CertificateError when producing/consuming using SASL_PLAIN on 1.4.5

### ISSUE THREAD

**Issue by jamoham** [dpkp/kafka-python#1770]
I was hitting Issue [#1666](https://github.com/dpkp/kafka-python/issues/1666)
So, I am looking to upgrade from 1.4.4 to 1.4.5

But in 1.4.5, I am getting the following exception when I try to produce/consume:
```
  File "/usr/local/lib/python2.7/site-packages/kafka/consumer/group.py", line 344, in __init__
    self._client = KafkaClient(metrics=self._metrics, **self.config)
  File "/usr/local/lib/python2.7/site-packages/kafka/client_async.py", line 230, in __init__
    self.config['api_version'] = self.check_version(timeout=check_timeout)
  File "/usr/local/lib/python2.7/site-packages/kafka/client_async.py", line 865, in check_version
    version = conn.check_version(timeout=remaining, strict=strict, topics=list(self.config['bootstrap_topics_filter']))
  File "/usr/local/lib/python2.7/site-packages/kafka/conn.py", line 990, in check_version
    if not self.connect_blocking(timeout_at - time.time()):
  File "/usr/local/lib/python2.7/site-packages/kafka/conn.py", line 321, in connect_blocking
    self.connect()
  File "/usr/local/lib/python2.7/site-packages/kafka/conn.py", line 413, in connect
    if self._try_handshake():
  File "/usr/local/lib/python2.7/site-packages/kafka/conn.py", line 484, in _try_handshake
    self._sock.do_handshake()
  File "/usr/local/Cellar/python@2/2.7.15_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/ssl.py", line 856, in do_handshake
    match_hostname(self.getpeercert(), self.server_hostname)
  File "/usr/local/Cellar/python@2/2.7.15_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/ssl.py", line 288, in match_hostname
    % (hostname, ', '.join(map(repr, dnsnames))))
ssl.CertificateError: hostname 'x.y.z.w' doesn't match either of 'host.name', '*.host.name'
```

I verified that in 1.4.5 the ssl library's `do_handshake` method, the IP address is being passed instead of the hostname. But in 1.4.4 the hostname is passed. That's why this issue is seen only in 1.4.5

Please let me know if you need any information.

I am using SASL_PLAIN auth and producing/consuming to port 9093.

**Comment by dpkp** [comment]
This has been fixed in master.

**Comment by jamoham** [comment]
@dpkp 
Thanks!
When will this fix be available in a release?

**Comment by jeffwidman** [comment]
I expect a release will happen once #1744 is resolved, and #1760 is confirmed fixed.
